l10n_fi_bank_barcode/models/account_move.py

```diff
@@ -95,10 +91,7 @@ def _compute_bank_barcode(self):
         for record in self:
 
             # Only EUR invoices are supported
-            # Since the default company currency is now USD, we have to bypass
-            # this check in the unit tests using the test_bank_barcode flag.
-            if record.company_currency_id.name != 'EUR' \
-                    and not self._context.get('test_bank_barcode'):
+            if record.currency_id.name != 'EUR':
```

Comment:
I've changed `company_currency_id` to `currency_id` because it makes more sense to me. Amount is encoded into barcode without currency code, therefore only EUR is supported as I get it.
Still, if we check `company_currency_id` we could have invoices in different currencies.